kotlinpoet/src/nonJvmMain/kotlin/com/squareup/kotlinpoet/Util.nonJvm.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.kotlinpoet
+
+internal actual fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> =
+  toMap()
+
+internal actual fun <T> Collection<T>.toImmutableList(): List<T> =
+  toList()
+
+internal actual fun <T> Collection<T>.toImmutableSet(): Set<T> =
+  toSet()
+
+internal actual fun Char.isJavaIdentifierStart(): Boolean {
+  return isLetter() ||
+    this in CharCategory.LETTER_NUMBER ||
+    this == '$' ||
+    this == '_'
+}
+
+internal actual fun Char.isJavaIdentifierPart(): Boolean {
+  //  TODO
+  //   A character may be part of a Java identifier if any of the following conditions are true:
```

Comment:
Should we maybe extract these checks into common to have consistent implementations on all platforms?

Reply:
Can be done in later PRs.

Reply:
Currently, I am dividing them because I am unsure if the current implementation of non-JVM is the same as the one in JVM.
It would be wonderful to merge them if we can confirm that they have the same effects!